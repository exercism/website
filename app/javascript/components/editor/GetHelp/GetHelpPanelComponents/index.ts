export { GetHelpPanelTrackHelp as TrackHelp } from './GetHelpPanelTrackHelp'
export { GetHelpPanelHints as Hints } from './GetHelpPanelHints'
export { GetHelpPanelCommunityHelp as CommunityHelp } from './GetHelpPanelCommunityHelp'
