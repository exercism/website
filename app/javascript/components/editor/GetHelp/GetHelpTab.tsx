import React from 'react'
import { Tab } from '../../common/Tab'
import { TabsContext } from '../../Editor'
import { GraphicalIcon } from '../../common/GraphicalIcon'

export const GetHelpTab = (): JSX.Element => (
  <Tab id="get-help" context={TabsContext}>
    <GraphicalIcon icon="help" />
    <span data-text="Get Help">Get help</span>
  </Tab>
)
