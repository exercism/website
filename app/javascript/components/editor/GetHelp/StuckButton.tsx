import React from 'react'
import { GraphicalIcon } from '@/components/common'
import { TabIndex } from '@/components/Editor'

type StuckButtonProps = {
  insider: boolean
  tab: TabIndex
  setTab: React.Dispatch<React.SetStateAction<TabIndex>>
}

export function StuckButton({
  insider,
  setTab,
  tab,
}: StuckButtonProps): JSX.Element {
  return (
    <button
      type="button"
      disabled={['get-help', 'chat-gpt'].includes(tab)}
      className="btn-enhanced btn-s !ml-0 mr-auto ask-chatgpt-btn"
      onClick={() => setTab(insider ? 'chat-gpt' : 'get-help')}
    >
      <GraphicalIcon icon="automation" height={16} width={16} />
      <span>Stuck? {insider ? 'Ask ChatGPT' : 'Get help'}</span>
    </button>
  )
}
