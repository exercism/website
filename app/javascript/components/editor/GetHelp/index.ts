export { GetHelpTab } from './GetHelpTab'
export { GetHelpPanel } from './GetHelpPanel'
