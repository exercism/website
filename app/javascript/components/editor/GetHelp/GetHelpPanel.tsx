import React from 'react'
import { TabsContext } from '@/components/Editor'
import { Tab } from '@/components/common'
import * as Component from './GetHelpPanelComponents'
import type { Assignment } from '../types'
import { Track } from '@/components/types'

export type GetHelpPanelProps = {
  helpHtml: string
  assignment: Assignment
  links: Record<'discordRedirectPath' | 'forumRedirectPath', string>
  track: Track
}

export function GetHelpPanel({
  helpHtml,
  assignment,
  links,
  track,
}: GetHelpPanelProps): JSX.Element {
  return (
    <Tab.Panel id="get-help" context={TabsContext}>
      <div className="pb-12 flex flex-col">
        <Component.Hints assignment={assignment} />
        <Component.CommunityHelp links={links} />
        <Component.TrackHelp helpHtml={helpHtml} track={track} />
      </div>
    </Tab.Panel>
  )
}
