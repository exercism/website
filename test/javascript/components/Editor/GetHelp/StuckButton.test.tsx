import React from 'react'
import { render, screen } from '@testing-library/react'
import { StuckButton } from '@/components/editor/GetHelp/StuckButton'

describe('Stuckbutton tests', () => {
  test('Component snapshot test', () => {
    const { asFragment } = render(
      <StuckButton insider={false} tab={'instructions'} setTab={() => null} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('Component shows correct button label if user is non-insider', () => {
    render(
      <StuckButton insider={false} tab={'instructions'} setTab={() => null} />
    )

    const label = screen.getByText('Stuck? Get help')
    expect(label).toBeInTheDocument()
  })
  test('Component shows correct button label if user is insider', () => {
    render(
      <StuckButton insider={true} tab={'instructions'} setTab={() => null} />
    )

    const label = screen.getByText('Stuck? Ask ChatGPT')
    expect(label).toBeInTheDocument()
  })

  test('Component is disabled if user is on the chat-gpt tab', () => {
    render(<StuckButton insider={true} tab={'chat-gpt'} setTab={() => null} />)

    const button = screen.getByRole('button')
    expect(button).toHaveAttribute('disabled')
  })
})
