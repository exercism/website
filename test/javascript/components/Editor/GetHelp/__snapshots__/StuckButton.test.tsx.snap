// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stuckbutton tests Component snapshot test 1`] = `
<DocumentFragment>
  <button
    class="btn-enhanced btn-s !ml-0 mr-auto ask-chatgpt-btn"
    type="button"
  >
    <img
      alt=""
      class="c-icon"
      height="16"
      src="undefined/assets/undefined"
      width="16"
    />
    <span>
      Stuck? Get help
    </span>
  </button>
</DocumentFragment>
`;
